evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -1418,6 +1482,49 @@ where
         }
     }
 
+    fn create_block_wrapper_circuit(
+        block: &BlockCircuitData<F, C, D>,
+    ) -> BlockWrapperCircuitData<F, C, D> {
+        let mut builder = CircuitBuilder::<F, D>::new(block.circuit.common.config.clone());
+
+        let parent_block_proof = builder.add_virtual_proof_with_pis(&block.circuit.common);
+        let parent_pv = PublicValuesTarget::from_public_inputs(&parent_block_proof.public_inputs);
+
+        let final_pv = add_virtual_final_public_values_public_input(&mut builder);
+
+        // This also enforces that the initial state trie root that will be stored in
+        // these `FinalPublicValues` actually matches the known checkpoint state trie
+        // root.
+        final_pv.connect_parent(&mut builder, &parent_pv);
+
+        let block_verifier_data = builder.constant_verifier_data(&block.circuit.verifier_only);
+
+        // We want these wrapped block proofs to have the exact same structure as 2-to-1
+        // aggregation proofs, so we add public inputs for cyclic verification,
+        // even though they'll be ignored.
+        let local_vk = builder.add_verifier_data_public_inputs();
+
+        builder.verify_proof::<C>(
+            &parent_block_proof,
+            &block_verifier_data,
+            &block.circuit.common,
+        );
+
+        // Pad to match the (non-existing yet!) 2-to-1 circuit's degree.
```

Comment:
It's unclear why `block.circuit.common.degree_bits()` matches the degree of the 2-to-1 circuit. Maybe explain that the value was found manually?

Reply:
It needs to be the same because the 2-to-1 circuit expects arbitrarily a wrapped proof or a 2-to-1 proof, hence they must have common degree

Reply:
Sorry, I meant: why is `block.circuit.common.degree_bits()` the right value, since we don't know the size of the 2-to-1 circuit yet?

Reply:
Ah sorry, I get the confusion now. Yes it should be `2-to-1` but because it doesn't exist yet, I felt slightly akward having a hardcoded integer lying around, so I used the previous circuit (as they end up being the same size anyway after padding). But maybe that ended up being more confusing indeed

Reply:
Ah sorry, I get the confusion now. Yes it should be `2-to-1` but because it doesn't exist yet, I felt slightly akward having a hardcoded integer lying around, so I used the previous circuit (as they end up being the same size anyway after padding). But maybe that ended up being more confusing indeed

Reply:
Yeah, that's why I was suggesting clarifying that the 2-to-1 circuit will end up having the same size of the block circuit after padding.

Reply:
Ok will mention it.